[Model Monitoring] Add _get_records() API for TimescaleDB connector

alxtkr77 opened the pull request:
## Description

Implements API parity between TimescaleDB and TDEngine connectors by adding `_get_records()` method to `TimescaleDBConnector`.

The TDEngine connector already has a `_get_records()` method that allows querying raw DataFrame data without specifying metrics upfront and supports querying all endpoints. This PR adds the same capability to TimescaleDB.

**Note**: `_get_records()` is an internal function used by tests and internal tooling only, not part of the public API.

## Ticket
Fixes: ML-11516

Commit:
[Model Monitoring] Add _get_records() API for TimescaleDB connector

Implement API parity with TDEngine by adding _get_records() method that allows
querying raw records without requiring metrics list upfront.

Changes:
- Update filter builders to accept None (no filtering)
- Make read_*_impl() methods accept Optional endpoint_id and metrics parameters
- Refactor read_predictions_impl() to support aggregation and be reused by read_predictions()
- Add _get_records() public method to TimescaleDBConnector
- Add comprehensive tests (6 tests, all passing)

Fixes: ML-11516

assaf758 commented on mlrun/model_monitoring/db/tsdb/timescaledb/timescaledb_connector.py:
```diff
@@ -289,6 +289,60 @@ def _enrich_mep_with_last_request(
+        columns: Optional[list[str]] = None,
+    ) -> pd.DataFrame:
+        """
+        Get raw records from TimescaleDB as pandas DataFrame (similar to TDEngine).
```
since we are going to remove TDEngine, I prefer if we don't mention it in TimeScaleDB

assaf758 commented on tests/model_monitoring/db/tsdb/timescaledb/test_timescaledb_get_records.py:
```diff
@@ -0,0 +1,227 @@
+
+
+class TestGetRecords:
+    """Tests for TimescaleDBConnector._get_records() method - API parity with TDEngine."""
```
again - remove TDEngine

assaf758 commented on tests/model_monitoring/db/tsdb/timescaledb/test_timescaledb_get_records.py:
```diff
@@ -0,0 +1,227 @@
+            == test_data[0][mm_schemas.MetricData.METRIC_NAME]
+        )
+
+    def test_get_records_results_all_endpoints(self, connector, query_test_helper):
```
please test the predictions table as well.

Commit:
[Model Monitoring] Address PR #8926 review comments

- Remove TDEngine references from _get_records() docstring
- Remove TDEngine reference from test class docstring
- Add test_get_records_predictions_all_endpoints() test

Addresses review feedback from @assaf758
